db_store/src/settings.rs

```diff
@@ -39,14 +39,37 @@ fn default_auth_type() -> AuthType {
 impl Settings {
     pub async fn connect(
         &self,
+        app_name: impl Into<String>,
         shutdown: triggered::Listener,
     ) -> Result<(Pool<Postgres>, futures::future::BoxFuture<'static, Result>)> {
         match self.auth_type {
             AuthType::Postgres => match self.simple_connect().await {
-                Ok(pool) => Ok((pool, Box::pin(async move { Ok(()) }))),
+                Ok(pool) => Ok((
+                    pool.clone(),
+                    Box::pin(metric_tracker::start(app_name.into(), pool, shutdown).await?),
```

Comment:
`metric_tracker::start` already returns a Box, so you're double boxing here. Either remove the box in the start function or here.

Reply:
Nice catch, Fixed 574c46a